src/IO/H5/CombineH5.hpp

```diff
@@ -6,8 +6,44 @@
 #include <string>
 #include <vector>
 
+#include "IO/Logging/Verbosity.hpp"
+
 namespace h5 {
 void combine_h5_vol(const std::vector<std::string>& file_names,
                     const std::string& subfile_name, const std::string& output,
                     bool check_src = true);
+
+/*!
+ * \brief Combine the `h5::Dat` subfiles of multiple `h5::H5File`s into a single
+ * H5 file.
+ *
+ * \details If there are overlapping times, the "latest" one is always used;
+ * meaning if you have data in `File1.h5` and `File2.h5` and if the first time
+ * in `File2.h5` is before some times in `File1.h5`, those times in `File1.h5`
+ * will be discarded and won't appear in the combined H5 file.
+ *
+ * If the H5 files in \p h5_files_to_combine have other types of subfiles, those
+ * will be ignored and will not appear in \p output_h5_filename. This function
+ * also assumes that the times in each of the \p h5_files_to_combine are already
+ * sorted.
```

Comment:
I think this second assumption is generally not valid for spectre, unfortunately. If you would like to keep it, I think you should add a check that it's true because someone will inevitably run this over a file where it's not true.